src/app/services/game-play.service.ts

```diff
@@ -83,6 +79,25 @@ export class GamePlayService {
     })
   }
 
+  public evaluateKey$(key: string): Observable<KeyScore> {
+    return combineLatest([this.answer$, this.acceptedGuesses$]).pipe(
+      map(([answer, acceptedGuesses]) => {
+        const correctLetters: string[] = acceptedGuesses.map(guess =>
+          guess.split('').filter((letter, idx) => letter === answer.word[idx])
+        ).flat();
+        const usedLetters: string[] = [...new Set(acceptedGuesses.join(''))];
+
+        if (!usedLetters.includes(key)) return KeyScore.unused;
+
+        if (correctLetters.includes(key)) return KeyScore.correct;
+
+        if (answer.word.includes(key)) return KeyScore.inWord;
+
+        return KeyScore.notInWord;
+      })
+    );
+  }
```

Comment:
Am I right that you want to call `evaluateKey$()` with a key, get a score back and then be done with that function/observable until the next time you call it?. If so, you might want to use `take(1)` inside the pipe. Otherwise `evaluateKey$()` returns an observable that keeps emitting whenever answer or acceptedGuesses emit.

Reply:
Yes that was the intention! I've been wondering when the take operator could come in handy so this is helpful to see.